test/balancerd/mzcompose.py

```diff
@@ -287,6 +419,31 @@ def workflow_mz_restarted(c: Composition) -> None:
     sql_cursor(c)
 
 
+def workflow_pgwire_param_rejection(c: Composition) -> None:
+    """Existing connections should fail if balancerd is restarted"""
+    c.up("balancerd", "frontegg-mock", "materialized")
+
+    def check_error(message: str, f: Callable[..., Any], expected_error: Any):
+        try:
+            f()
+        except Exception as e:
+            assert isinstance(e, expected_error)
+            return
+        raise AssertionError(f"Expected {message} to raise {expected_error}")
+
+    check_error(
+        "connect with x_forwarded_for param",
+        lambda: sql_cursor(c, init_params={"mz_forwarded_for": "1.1.1.1"}),
+        InterfaceError,
+    )
+
+    check_error(
+        "connect with mz_connection_id param",
+        lambda: sql_cursor(c, init_params={"mz_connection_uuid": "123456"}),
+        InterfaceError,
+    )
```

Comment:
Oh wow, thanks for adding this! `mz_connection_id` isn't going to be used for anything as load bearing as `x_forwarded_for` so it didn't feel essential to test it when we added it, but nonetheless very glad to have a test for it!